src/components/button/index.ts

```diff
@@ -0,0 +1,26 @@
+import Block from "../../utils/Block";
+import template from "./button.pug";
+import "./styles.css";
+
+interface ButtonProps {
+  label: string;
+  events?: {
+    click: () => void;
+  };
+  classes?: string;
+  url?: string;
+  type?: string;
```

Comment:
Для таких данных, как тип чего-то, ```string``` не очень подходит. Тип - это ограниченное количество значений. ```string``` допускает любые значения. Тут лучше подойдет использование union'a
```suggestion
  type?: "button" | "submit";
```